src/Compatibility/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -43,7 +44,7 @@ public SuppressionEngine(string? suppressionsFile = null, string ? noWarn = null
         /// <param name="left">Optional. The left operand in a APICompat error.</param>
         /// <param name="right">Optional. The right operand in a APICompat error.</param>
         /// <returns><see langword="true"/> if the error is already suppressed. <see langword="false"/> otherwise.</returns>
-        public bool IsErrorSuppressed(string diagnosticId, string? target, string? left = null, string? right = null, bool isBaselineSuppression = false)
+        public bool IsErrorSuppressed(string diagnosticId, string? target = null, string? left = null, string? right = null, bool isBaselineSuppression = false)
```

Comment:
I think we actually want diagnosticId to be nullable in order to fully support a roundtrip. Here are my thoughts on this: What happens if a customer decides to to a global suppression between a left and a right pair (meaning he doesn't use diagnosticID) but then wants to run the engine again and regenerate the suppression file? If I'm understanding correctly, with these changes a diagnosticId property will be created with string.Empty which is not desired.